test/parser.js

```diff
@@ -0,0 +1,313 @@
+'use strict';
+
+const metatests = require('metatests');
+const parser = require('../lib/parser');
+const { writeBigIntToBuffer } = require('../lib/bigint-buffer');
+
+const LONG_BUFFER_SIZE = 16384;
+
+
+const fillHandskakeBuffer = (handshake, buf) => {
+  buf.writeIntLE(parser.constants.PROTOCOL_VERSION, 0, 2);
+  buf.writeIntLE(handshake.status, 2, 1);
+  buf.writeIntLE(handshake.reserved, 3, 1);
+  buf.write(handshake.token, 4);
+};
+
+const fillParcelBuffer = (parcel, buf) => {
+  buf.writeIntLE(parser.constants.STRUCT_PARCEL, 1);
+  buf.writeIntLE(parcel.parcelId, 1, 4);
+  buf.writeIntLE(parcel.parcelType, 5, 1);
+  buf.writeIntLE(parcel.compression, 6, 1);
+  buf.writeIntLE(parcel.encoding, 7, 1);
+  writeBigIntToBuffer(parcel.length, buf, 8);
+};
+
+const fillChunkBuffer = (chunk, buf) => {
+  buf.writeIntLE(parser.constants.STRUCT_CHUNK, 0, 1);
+  buf.writeIntLE(chunk.parcelId, 1, 4);
+  buf.writeIntLE(chunk.chunkId, 5, 4);
+  buf.writeIntLE(chunk.flag, 9, 1);
+  buf.writeIntLE(chunk.length, 10, 2);
+  chunk.payload.copy(buf, 12);
+};
+
+const handshakeTestData = () => {
```

Comment:
If you don't mutate data it will be better to put all this test data to separate object and initialize them all in dedicated functions.